### DISCUSSION TITLE
Can't connect to WiFi

### DISCUSSION THREAD

**Discussion by Apricot63** [micropython/micropython#12178] [ESP8266] [answered]
Hello,

I can't connect to my WiFi network if i use prepared script - "main.py".  It is possible to connect, if I type commands one by one through the Thonny console (ESP-12F as an interpreter ofc).

I use Thonny as IDE to save scripts into ESP-12F file system via CH340G serial-USB converter. Firmware: esp8266-20230426-v1.20.0.bin

boot.py
```
import uos, machine
import gc
gc.collect()
```

main.py
```
import network
import machine
print("\n\n#############\n")
sta = network.WLAN(network.STA_IF)
sta.active(True)
print(sta.scan())
sta.connect("banana", "vExolpDFG")
for i in range(10) :
    machine.lightsleep(100)
    if sta.isconnected() :
        break
    print(i)
    print(sta.isconnected())
    sta.connect("banana", "vExolpDFG")
if sta.isconnected() :
    sta.disconnect()
```

CONSOLE OUTPUT:
```
#############
#5 ets_task(4020f544, 28, 3fff9ee8, 10)

[SSIDS ARE HERE]
0
False
1
False
####### (...)
False
9
False

MicroPython v1.20.0 on 2023-04-26; ESP module with ESP8266
Type "help()" for more information.
>>> 
```

Windows hotspot service, tells me esp is already connected
![image](https://github.com/micropython/micropython/assets/82253964/deaf4c26-03cd-44c4-b6b6-c7b83b5dd3e7)

**Comment by davefes** [comment, marked as answer]
I'd remove the first:
```py
sta.connect("banana", "vExolpDFG")
```
then I'd copy the example in:
https://docs.micropython.org/en/latest/esp8266/quickref.html

> A useful function for connecting to your local WiFi network is:

Sometimes it is best to start with a:
```py
sta.active(False)
```
For a bullet-proof WiFi connect look at:
https://github.com/glenn20/micropython-espnow-utils/blob/main/src/wifi.py

**Reply by Apricot63** [reply]
Thank you :)
I used defined function from examples and it works! - I added at the beginning few lines of code - for sure :)
```
def do_connect():
    import network
    wlan = network.WLAN(network.AP_IF)
    wlan.active(False)
(...)
```
Bullet-proof WiFi libary will be helpful as well in future...

**Comment by robert-hh** [comment]
I would remove the sta.connect() statement in the loop and NOT use machine.lightsleep(). Use time.sleep_me() instead, even if looking at the code of lightsleep() it does not make a difference.

**Comment by mnvoronin** [comment]
Do not use machine.lightsleep() because it shuts down WiFi. Use time.sleep_ms() instead.
